### PR TITLE
[hostdenied/fresh-backup] Reverse order of series so new entries are …

### PR DESCRIPTION
**Pull request by shtrom**
…on top

Signed-off-by: Olivier Mehani <shtrom@ssji.net>

### CODE REVIEW

File: plugins/backup/fresh-backups [hunk 1]
```diff
@@ -76,19 +76,22 @@ graph_title Fresh backups
...
-	cat << EOF
-age${AGE}.label older than $((AGE*24))h
-age${AGE}.draw STACK
+for AGE in $(seq "${LIFETIME}" -1 0); do
+	if [ "${AGE}" = 0 ]; then
+		echo "age${AGE}.label today"
+	else
+		echo "age${AGE}.label older than $((AGE*24))h"
+	fi
+	if [ "${AGE}" = "${LIFETIME}" ]; then
```
Comment (sumpfralle) [inline plugins/backup/fresh-backups]: You can simplify this with using `AREASTACK` for all fields.

File: plugins/backup/fresh-backups [hunk 2]
```diff
@@ -76,19 +76,22 @@ graph_title Fresh backups
...
+		echo "age${AGE}.draw AREA"
+	else
+		echo "age${AGE}.draw STACK"
+	fi
+		cat << EOF
```
Comment (sumpfralle) [inline plugins/backup/fresh-backups]: The indentation can be reduced, I guess.

File: plugins/backup/fresh-backups [hunk 3]
```diff
@@ -76,19 +76,22 @@ graph_title Fresh backups
...
 age${AGE}.colour ${COLOUR[$AGE]}
-age${AGE}.warning :
-age${AGE}.critical :
+age${AGE}.warning
```
Comment (sumpfralle) [inline plugins/backup/fresh-backups]: Do these two lines serve a purpose?
How about using `print_warning` instead?

File: plugins/ssh/hostdenied [hunk 4]
```diff
@@ -44,35 +44,38 @@
...
-NAME="$(basename $0)"                                        # component of naming temporary files
+NAME="$(basename "$0")"                                        # component of naming temporary files
 STATEFILE="$MUNIN_PLUGSTATE/$NAME.state"
 COLOUR=(FF0000 DA0024 B60048 91006D 6D0091 4800B6 2400DA 0000FF)  # hot to cold colours
 # ------------------------------------------------------------------------------------------------------
 
 run_autoconf()  {
-        RUN="no"
-        which grep denyhosts basename > /dev/null && RUN="yes"             # only run when grep and denyhost are present
+	RUN="no (denyhosts not found)"
```
Comment (sumpfralle) [inline plugins/ssh/hostdenied]: The indentation is mixed.

File: plugins/ssh/hostdenied [hunk 5]
```diff
@@ -44,35 +44,38 @@
...
-age${AGE}.draw STACK
-age${AGE}.colour ${COLOUR[$AGE]}
 EOF
+for AGE in {7..0}; do
+	if [ "${AGE}" = 0 ]; then
+		echo "age${AGE}.label added last 24h"
+	else
+		echo "age${AGE}.label older than $((AGE*24))h"
+	fi
+	if [ "${AGE}" = 7 ]; then
```
Comment (sumpfralle) [inline plugins/ssh/hostdenied]: Maybe `AREASTACK`?

File: plugins/backup/fresh-backups (hunk 3 above)
Comment (shtrom) [reply plugins/backup/fresh-backups]: In my early experiment with this, the warning and critical values seemed to be inherited from those for `freshcount` when plotted. This forces no limit to be checked on those series.

`plugin.sh` is not safe to use with `-u` as it doesn't default undefined values (such as warnings and critical limits) to sane values.

**Comment by shtrom** [comment]
@sumpfralle done.

File: plugins/backup/fresh-backups (hunk 3 above)
Comment (sumpfralle) [reply plugins/backup/fresh-backups]: Inheritance: interesting. I am not sure, that this is the way it should work ...

`set -u`: indeed, this needs to be fixed!

**Comment by sumpfralle** [comment]
Thank you!

**Pull request merged**